src/events/pg.js

```diff
@@ -0,0 +1,101 @@
+const uuid4 = require('uuid4');
+const shimmer = require('shimmer');
+const pg = require('pg')
+const sqlParser = require('node-sqlparser');
+const utils = require('../utils.js');
+const tracer = require('../tracer.js');
+const serverlessEvent = require('../proto/event_pb.js');
+const eventInterface = require('../event.js');
+const errorCode = require('../proto/error_code_pb.js');
+const config = require('../config.js');
+
+function pgClientWrapper(wrappedFunction) {
+    return function internalpgClientWrapper(queryString, arg1, arg2) {
+        const callback = (arg2 === undefined) ? arg1 : arg2;
+        const params = (arg1 instanceof Function) ? [] : arg1;
+
+        const sqlObj = sqlParser.parse(queryString);
+        let databaseType = 'local';
+        const {
+            type,
+            columns,
+            table,
+            from
+        } = sqlObj;
+
+        const {
+            database,
+            host
+        } = this.connectionParameters;
+
+        if (host.match('.rds.')) { databaseType = 'rds' }
+        if (host.match('.redshift.')) { databaseType = 'redshift' }
+
+        let clientRequest = null;
+
+        try {
+            const resource = new serverlessEvent.Resource([
+                database, // name of the database
+                databaseType,
+                queryString
+            ]);
+
+            const startTime = Date.now();
+
+            const dbapiEvent = new serverlessEvent.Event([
+                `dbapi-${uuid4()}`,
+                utils.createTimestampFromTime(startTime),
+                null,
+                'pg',
+                0,
+                errorCode.ErrorCode.OK,
+            ]);
+
+            dbapiEvent.setResource(resource);
+
+            let extendedData = {};
+            if ((type == 'select') || (!tracer.metadata_only)) {
+                extendedData['Query'] = queryString
+            }
+
+            eventInterface.addToMetadata(dbapiEvent, {
+                'Host': host,
+                'Driver': 'pg',
+                'Table Name': table
+            }, extendedData);
+            let patchedCallback = undefined;
+
+            const responsePromise = new Promise((resolve) => {
+                patchedCallback = (err, res) => {
+                    if (callback) {
+                        callback(res);
+                    }
+                    resolve();
+                };
+            }).catch((err) => {
+                tracer.addException(err);
+            });
+
+            responsePromise.then(() => {
+                dbapiEvent.setDuration(utils.createDurationTimestamp(startTime));
```

Comment:
Is there anything else you can gather from the response? maybe returned count of rows/value?